mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,384 @@
+
+#. Add the following ``ConditionTagHelper`` class to the *TagHelpers* folder.
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/ConditionTagHelper.cs
+    :language: c#
+    
+2. Replace the contents of the *Views/Home/Index.cshtml* file with the following markup:
+
+.. literalinclude::  authoring/sample/TagHlp/src/AuthoringTagHelpers/Views/Home/Index.cshtml
+   :language: aspx-cs
+
+3. Replace the ``Index`` method in the ``Home`` controller with the following code:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/Controllers/HomeController.cs
+   :language: c#
+   :lines: 9-18
+   :dedent: 6
+   
+4. Run the app and browse to the home page. The markup in the conditional ``div`` will not be rendered. Append  the query string ``?approved=true`` to the URL (for example, http://localhost:1235/Home/Index?approved=true). The approved is set to true and the conditional markup will be displayed.
+
+Note: We use the `nameof <https://msdn.microsoft.com/en-us/library/dn986596.aspx>`_ operator to specify the attribute to target rather than specifying a string as we did with the bold tag helper:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/zConditionTagHelperCopy.cs
+    :language: c#
+    :lines: 6-19
+    :emphasize-lines: 1,2,5
+
+The `nameof <https://msdn.microsoft.com/en-us/library/dn986596.aspx>`_ operator will protect the code should it ever be refactored (we might want to change the name to RedCondition).
+
+Avoiding Tag Helper conflicts
+______________________________
+
+In this section, we will write a pair of auto-linking tag helpers. The first will replace markup containing a URL starting with HTTP to an HTML anchor tag containing the same URL (and thus yielding a link to the URL). The second will do the same for a URL starting with WWW.
+
+Because these two helpers are closely related and we may refactor them in the future, we'll keep them in the same file.
+
+#. Add the following ``AutoLinker`` class to the *TagHelpers* folder. 
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/z1AutoLinker.cs
+   :language: c#
+   :lines: 7-20
+   :dedent: 3
+
+Notes: The ``AutoLinkerHttpTagHelper`` class targets ``p`` elements and uses `Regex <https://msdn.microsoft.com/en-us/library/system.text.regularexpressions.regex.aspx>`__ to create the anchor.
+
+
+2. Add the following markup to the end of the *Views/Home/Contact.cshtml* file:
+
+.. literalinclude::  authoring/sample/TagHlp/src/AuthoringTagHelpers/Views/Home/Contact.cshtml
+   :language: aspx-cs
+   :lines: 1-20 
+   :emphasize-lines: 20
+
+3. Run the app and verify that the tag helper renders the anchor correctly.
+
+4. Update the ``AutoLinker`` class to include the ``AutoLinkerWWWTagHelper`` which will convert www text to an anchor tag that also contains the original www text. The updated code is highlighted below:
```

Comment:
Will need to update the name instances of `AutoLinkerWWWTagHelper` after the rename.